lib/private/RichObjectStrings/Validator.php

```diff
@@ -16,30 +18,22 @@
  * @since 11.0.0
  */
 class Validator implements IValidator {
-	/** @var Definitions */
-	protected $definitions;
-
-	/** @var array[] */
-	protected $requiredParameters = [];
+	protected array $requiredParameters = [];
 
-	/**
-	 * Constructor
-	 *
-	 * @param Definitions $definitions
-	 */
-	public function __construct(Definitions $definitions) {
-		$this->definitions = $definitions;
+	public function __construct(
+		protected Definitions $definitions,
+	) {
 	}
 
 	/**
 	 * @param string $subject
-	 * @param array[] $parameters
+	 * @param array<non-empty-string, array> $parameters
```

Comment:
```suggestion
	 * @param array<non-empty-string, array<non-empty-string, non-empty-string>> $parameters
```
Is this correct? (same for the interface

Reply:
Right, I attempted to "enforce" the `array{type: string, id: string, name: string}` but since we don't allow additional elements I failed to finish that and went back to array.
But totally makes sense to at least require strings

Reply:
It's potentially empty though:
https://github.com/nextcloud/server/blob/3a6716895a66b2f4589f47f114b1cdad19198b1f/lib/public/RichObjectStrings/Definitions.php#L317-L322